tools/prepare-release.sh

```diff
@@ -0,0 +1,232 @@
+    say "Tag description for v$version:"
+    echo "$tag_text"
+    git tag -a v"$version" "$branch" -m "$tag_text"
+    say "Local tag v$version created."
+}
+
+function cmd_help {
+    cat <<EOF
+$0 <version>
+
+    Example: ./tools/prepare-release.sh 0.42.0
+
+    Prepare a new Firecracker release:
+    1. Update the version number
+    2. Update Crate dependencies
+    3. Generate CREDITS.md and CHANGELOG.md
+    4. Commit the result
+    5. Tag with "git tag"
+EOF
+}
+
+# # # # MAIN # # # #
+
+if [ $# -ne 1 ]; then
+    cmd_help;
+    exit 1;
+fi
+version=$1
+validate_version "$version"
+
+ORIGIN_URL=$(git config --get remote.origin.url)
+REPO=$(basename "$ORIGIN_URL" .git)
+GH_USER=$(git config --get github.user)
+LOCAL_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+UPSTREAM=upstream
+RELEASE_BRANCH=firecracker-v$(echo "$version" |cut -d. -f-2)
+if [ "$LOCAL_BRANCH" != "$RELEASE_BRANCH" ]; then
+    cat <<EOF
+ERROR: The current branch '$LOCAL_BRANCH' should be '$RELEASE_BRANCH'
+Bailing out.
+You can fix this by checking out the expected release branch and rerun:
+
+    git checkout -B $RELEASE_BRANCH
+EOF
+    exit 1
+fi
+
+# In which branch should the change go, in the main repo?
+TARGET_BRANCH=main
+PATCH=$(echo "$version" |cut -d. -f3)
+# If this is a patch release, the target branch should be the release branch
+if [ "$PATCH" -gt 0 ]; then
+    TARGET_BRANCH=$RELEASE_BRANCH
+fi
+PR_URL="https://github.com/firecracker-microvm/$REPO/compare/$TARGET_BRANCH...$GH_USER:$REPO:$LOCAL_BRANCH?expand=1"
+
+# Get current version from the swagger spec.
+swagger="$FC_ROOT_DIR/src/api_server/swagger/firecracker.yaml"
+prev_ver=$(get_swagger_version "$swagger")
+
+say "Updating from $prev_ver to $version ..."
+
+# Update version in files.
+files_to_change=(
+    "$swagger"
+    "$FC_ROOT_DIR/src/firecracker/Cargo.toml"
+    "$FC_ROOT_DIR/src/jailer/Cargo.toml"
+    "$FC_ROOT_DIR/src/rebase-snap/Cargo.toml"
+    "$FC_ROOT_DIR/src/seccompiler/Cargo.toml"
+)
+say "Updating source files:"
+for file in "${files_to_change[@]}"; do
+    say "- $file"
+    # Dirty hack to make this work on both macOS/BSD and Linux.
+    # FIXME This is very hacky and can unintentionally bump other versions, so
+    # only do the replacement *once*.
+    sed -i "s/$prev_ver/$version/" "$file"
+done
+
+# Run `cargo check` to update firecracker and jailer versions in
+# `Cargo.lock`.
+say "Updating lockfile..."
+cargo check
+
+# Update credits.
+say "Updating credits..."
+$FC_TOOLS_DIR/update-credits.sh
+
+# Update changelog.
+say "Updating changelog..."
+sed -i "s/\[Unreleased\]/\[$version\]/g" "$FC_ROOT_DIR/CHANGELOG.md"
+
+git add "${files_to_change[@]}" Cargo.lock CREDITS.md CHANGELOG.md
+git commit -s -m "Releasing v$version"
+
+# Start by creating a local tag and associate to it a description.
+say "Creating local tag..."
+create_local_tag "$version" "$prev_ver" "$LOCAL_BRANCH" || die "Could not create local tag v$version."
```

Comment:
I am not sure we should do this step here. 
We want to do this after the PR with the changelog update has been merged (i.e as part of the actual `make_release` step) since the changelog may suffer changes after PR review and the tag description would not have them, right?

Reply:
hmm true. We could:

1. Separate that step as `release-tag.sh`?
2. Provide instructions to do the tag by hand

I would probably go with 1). 2) seems too error prone.

Reply:
Yes, let s go with 1.